app_check/integration_test/src/integration_test.cc

```diff
@@ -157,6 +167,8 @@ void FirebaseAppCheckTest::TerminateAppCheck() {
       delete app_check;
     }
   }
+
+  firebase::app_check::AppCheck::SetAppCheckProviderFactory(nullptr);
```

Comment:
It looks like this might be problematic for android as installAppCheckProviderFactory expects a non-null param
https://github.com/firebase/firebase-android-sdk/blob/master/appcheck/firebase-appcheck/src/main/java/com/google/firebase/appcheck/FirebaseAppCheck.java#L54

But I do think we want some way to reset app check between tests, so I'll worry about that while I am working on android implementation, and this line of code is fine for this PR.

Reply:
Hmm, yeah.  I think having some way to clear it, especially for tests, would be good.  If it can't be done on Android, we can figure out another plan though.